aviatrix/resource_aviatrix_remote_syslog.go

```diff
@@ -0,0 +1,161 @@
+				Default:      "TCP",
+				ValidateFunc: validation.StringInSlice([]string{"TCP", "UDP"}, false),
+				Description:  "Protocol: TCP or UDP (TCP by default)",
+			},
+			"template": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Optional Custom Template: Useful when forwarding to 3rd party servers like Datadog or Sumo",
+			},
+			"exclude_gateway": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "List of excluded gateways.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"notls": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if not protected by TLS.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Enabled or not.",
+			},
+		},
+	}
+}
+
+func resourceAviatrixRemoteSyslogCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	remoteSyslog := &goaviatrix.RemoteSyslog{
+		Server:            d.Get("server").(string),
+		Port:              d.Get("port").(int),
+		Protocol:          d.Get("protocol").(string),
+		Index:             d.Get("index").(int),
+		Template:          d.Get("template").(string),
+		CaCertificate:     d.Get("ca_certificate_file_path").(string),
+		PublicCertificate: d.Get("public_certificate_file_path").(string),
+		PrivateKey:        d.Get("private_key_file_path").(string),
+	}
+
+	if !((remoteSyslog.CaCertificate != "" && remoteSyslog.PublicCertificate != "" && remoteSyslog.PrivateKey != "") ||
+		(remoteSyslog.CaCertificate == "" && remoteSyslog.PublicCertificate == "" && remoteSyslog.PrivateKey == "")) {
+		return fmt.Errorf("one or more certificates missing")
+	}
+
+	var excludeGateways []string
+	for _, v := range d.Get("exclude_gateway").(*schema.Set).List() {
+		excludeGateways = append(excludeGateways, v.(string))
+	}
+	if len(excludeGateways) != 0 {
+		remoteSyslog.ExcludeGatewayInput = strings.Join(excludeGateways, ",")
+	}
+
+	if err := client.EnableRemoteSyslog(remoteSyslog); err != nil {
+		return fmt.Errorf("could not enable remote syslog: %v", err)
+	}
+
+	d.SetId("remote_syslog_" + strconv.Itoa(remoteSyslog.Index))
+	return nil
+}
+
+func resourceAviatrixRemoteSyslogRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	remoteSyslogStatus, err := client.GetRemoteSyslogStatus(d.Get("index").(int))
+	if err != nil {
+		return fmt.Errorf("could not get remote syslog status: %v", err)
+	}
+
+	d.Set("notls", remoteSyslogStatus.Notls)
+	d.Set("status", remoteSyslogStatus.Status)
+
+	var excludedGateways []interface{}
+	for _, v := range remoteSyslogStatus.ExcludedGateway {
+		excludedGateways = append(excludedGateways, v)
+	}
+	if err := d.Set("exclude_gateway", excludedGateways); err != nil {
+		return fmt.Errorf("could not set excluded_gateway: %v", err)
+	}
+
+	d.SetId("remote_syslog_" + remoteSyslogStatus.Index)
+	return nil
+}
+
```

Comment:
Your `Read` function needs to `d.Set` every attribute except for the file path attributes since you cannot retrieve those values from the API. 

Also, in the `Read` we need to handle the case of `import` this current implementation failed to import correctly for me. Check out some other resource's `Read` functions, they will often have some logic at the top of the function to handle importing.